src/app/ui/cpwsarea/wsnewmap/wsnewmap.go

```diff
@@ -0,0 +1,199 @@
+package wsnewmap
+
+import (
+	"log"
+	"math"
+	"path/filepath"
+	"sdmm/app/command"
+	"sdmm/app/prefs"
+	"sdmm/app/ui/cpwsarea/workspace"
+	"sdmm/app/window"
+	"sdmm/dmapi/dmenv"
+	"sdmm/imguiext"
+	"sdmm/imguiext/icon"
+	"sdmm/imguiext/markdown"
+
+	"github.com/SpaiR/imgui-go"
+	"github.com/sqweek/dialog"
+)
+
+type App interface {
+	LoadedEnvironment() *dmenv.Dme
+
+	CommandStorage() *command.Storage
+	Prefs() prefs.Prefs
+
+	DoOpenMapByPath(path string)
+	DoClose()
+}
+
+type WsNewMap struct {
+	workspace.Content
+
+	app App
+
+	mapWidth  int // x
+	mapHeight int // y
+	mapZDepth int // z
+
+	multiZType string
+
+	newMapMetaFile string
+}
+
+func New(app App) *WsNewMap {
+	return &WsNewMap{
+		app: app,
+
+		mapWidth:       1,
+		mapHeight:      1,
+		mapZDepth:      1,
+		multiZType:     Default,
+		newMapMetaFile: "",
+	}
+}
+
+func (ws *WsNewMap) Name() string {
+	return icon.FaPlus + " New Map"
+}
+
+func (ws *WsNewMap) Title() string {
+	return "New Map"
+}
+
+func (ws *WsNewMap) Process() {
+	ws.showContent()
+}
+
+func (ws *WsNewMap) showContent() {
+	// Header
+	markdown.ShowHeader("Create a new Map", window.FontH2)
+	imgui.Separator()
+	imgui.NewLine()
+
+	// Width Height
+	markdown.ShowHeader("Width (X)", window.FontH3)
+
+	imgui.PushTextWrapPos()
+	imgui.TextDisabled("The width of the map. Minimum is 1.")
+	imgui.PopTextWrapPos()
+
+	vwidth := int32(ws.mapWidth)
+	if imguiext.InputIntClamp("##map_width", &vwidth, 1, math.MaxInt32, 1, 1) {
+		if int(vwidth) != ws.mapWidth {
```

Comment:
This is fine just to set the field anywhere. Even if it's the same. To avoid unneeded `if`s.